config/Cargo.toml

```diff
@@ -16,3 +21,6 @@ apache-avro = { version = "0.14.0", features = ["snappy"] }
 
 [dev-dependencies]
 pretty_assertions = "1.3.0"
+
+[features]
+calculate-page-capacity = []
```

Comment:
How do you think about maybe putting all these calculation logic in `bench` folder? Since this seems like a benchmark if we do that then I think we can get away with having an extra feature for this. In general I'm a bit conservative towards using features unless absolutely necessary (like for wasm) and prefer alternative approaches.

Reply:
We can certainly separate the code out into a separate folder; however, the bigger issue is I wanted to be able to run the graph calculation without having to run all the other tests. But the important part is not running the graph calculation every time we run tests, which is achieved with only the one `cfg` statement around the graph calculation module. I say we leave the rest of these for now; if it gets messy to maintain we can remove them later.

I initially tried to set up the graph capacity calculation as an actual benchmark, but apparently the `#[bench]` attribute is currently unstable & only available in nightly. So I had to resort to setting it up as a test.

The other alternative would be to adopt the Substrate approach to benchmarking, which would be to build a binary and give it a CLI mode to run the calculation. But that seemed like overkill.

Reply:
Another alternative approach might be yo just `ignore` that benchmarking test just like parity does https://github.com/paritytech/substrate/blob/master/frame/message-queue/src/integration_test.rs#L125
I think that would be cleaner compared to adding feature flag on all test location. Any thoughts?

Reply:
The issue isn't how to ignore the benchmarking test--that's easy and only requires an attribute in one spot. The issue is I want to be able to ignore all the other tests when we want to run the benchmark. To do that we need to flag all the other test modules. Unless you thought of an easier way?

One other option would be to forget about all that, and just control it at the Makefile level like this:

```
calculate-page-capacities:
     cargo test --feature=calculate_page_capacities calculate_page_capacities
```

The last argument is a string matcher for the test(s) to run.

If you prefer that, let me know; it's easy enough to change.

Reply:
I honestly prefer the last mentioned option compared to adding feature flags to all tests. The reason is that this is just needs to be ran every once in a while if not only once.